src/main/java/faang/school/projectservice/validator/MomentValidator.java

```diff
@@ -0,0 +1,39 @@
+package faang.school.projectservice.validator;
+
+import faang.school.projectservice.dto.moment.MomentDto;
+import faang.school.projectservice.exception.DataValidationException;
+import faang.school.projectservice.model.Project;
+import faang.school.projectservice.model.ProjectStatus;
+import faang.school.projectservice.repository.ProjectRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class MomentValidator {
+    private final ProjectRepository projectRepository;
+
+    public void validateMoment(MomentDto momentDto) {
+        validateProjectIds(momentDto);
+
+    }
+
+    public void validateProjectIds(MomentDto momentDto) {
+        if (momentDto.getProjectIds().isEmpty()) {
+            throw new DataValidationException("Empty projectIds");
+        } else {
+            for (long projectId : momentDto.getProjectIds()) {
```

Comment:
it is quite resource -intensive to establish connection with a data base every single iteration, so it is better to use findAllByIds